ios/fastlane/Fastfile

```diff
@@ -15,164 +15,206 @@
-
-    push_to_git_remote(remote_branch: 'master')
-
-    build_app(
-      scheme: 'ReactNativeBase-Prod',
-      workspace: workspace,
-      include_bitcode: true,
-      export_method: 'app-store'
+  lane :debug_develop do
+    build_and_sign(
+      type: 'adhoc',
+      scheme: app_name+'-Develop',
+      method: 'ad-hoc'
     )
+  end
 
-    changelog_from_git_commits(
-      pretty: "- (%ae) %s",# Optional, lets you provide a custom format to apply to each commit when generating the changelog text
-      date_format: "short",# Optional, lets you provide an additional date format to dates within the pretty-formatted string
-      match_lightweight_tag: false,  # Optional, lets you ignore lightweight (non-annotated) tags when searching for the last tag
-      merge_commit_filtering: "exclude_merges" # Optional, lets you filter out merge commits
+  lane :debug_staging do
+    build_and_sign(
+      type: 'appstore',
+      scheme: app_name+'-Staging',
+      method: 'app-store'
     )
-
-    upload_to_testflight(username: username)
   end
-
-  lane :build_develop do
-    sync_code_signing(
-      git_url: certificates_git_url,
-      username: username,
-      type: 'appstore'
+  
+  lane :debug_production do
+    build_and_sign(
+      type: 'appstore',
+      scheme: app_name,
+      method: 'app-store'
     )
+  end
 
-    increment_build_number(xcodeproj: xcodeproj)
-
-    build_app(
-      scheme: 'ReactNativeBase-Develop',
-      workspace: workspace,
-      include_bitcode: true,
-      export_method: 'app-store'
+  # Generate AdHoc build and push to S3 - modify scheme as appropiate
+  lane :release_qa_adhoc do  
+    
+    #ensure_git_status_clean
+    build_and_sign(
+      scheme: app_name+'-Develop',
+      type: 'ad-hoc'
+    )
+    # Push .ipa and metadata to S3
+    aws_s3(
+      access_key: s3_key,
+      secret_access_key: s3_secret_key,
+      bucket: s3_bucket,
+      region: s3_region,
+      ipa: lane_context[SharedValues::IPA_OUTPUT_PATH],
+      path: folder+"/ios/"+branch+"-{CFBundleShortVersionString}.{CFBundleVersion}/",
+      acl: "private",
+      upload_metadata: true
+    )
+    slack(
+      message: "Hi! A new iOS "+app_name+"-Develop build has been pushed to S3",
+      payload: {
+        "Build Date" => Time.new.to_s,
+        "Build Output" => "https://github.com/rootstrap/"+folder+"/actions",
+        "S3 location" => "s3://"+s3_bucket+"/"+folder+"/ios/"+branch+"-"+version+"."+build,
+        "HTTP link" => lane_context[SharedValues::S3_IPA_OUTPUT_PATH]
+      },
+      channel: slack_channel,
+      slack_url: slack_url,
+      use_webhook_configured_username_and_icon: true,
+      fail_on_error: false,
+      success: true
     )
   end
 
-  lane :build_staging do
-    sync_code_signing(
-      git_url: certificates_git_url,
-      username: username,
-      type: 'appstore'
+  # Generate Appstore build and push to TestFlight - modify scheme as appropiate
+  lane :release_qa_appstore do
```

Comment:
same comment I had on the android file. I think that it is worth having a separate build target for QA if that is sth that we are aiming to have for release in the base.

Dev build target and release can also be useful to set up since some times there are some issues that only happen when testing via testflight or google play console and it is best to have an app that you can update and not break what QA or staging are testing.

Reply:
gotcha. Will add Dev releases and restructure the fastfiles a bit to minimize duplicated code